modules/azure/billing-alert-functions/main.tf

```diff
@@ -0,0 +1,171 @@
+}
+
+resource "azurerm_resource_group" "main" {
+  count    = var.to_provision == true ? 1 : 0
+  name     = "rg-${var.func_identifier}"
+  location = var.location
+}
+
+resource "azurerm_storage_account" "func" {
+  count                    = var.to_provision == true ? 1 : 0
+  name                     = format("st%s", replace(var.func_identifier, "-", ""))
+  resource_group_name      = azurerm_resource_group.main[count.index].name
+  location                 = azurerm_resource_group.main[count.index].location
+  account_tier             = var.storage.tier
+  account_replication_type = var.storage.replication_type
+  min_tls_version          = "TLS1_2"
+
+  queue_properties {
+    logging {
+      delete                = true
+      read                  = true
+      write                 = true
+      version               = "1.0"
+      retention_policy_days = 7
+    }
+
+    hour_metrics {
+      enabled               = true
+      include_apis          = true
+      version               = "1.0"
+      retention_policy_days = 7
+    }
+
+    minute_metrics {
+      enabled               = true
+      include_apis          = true
+      version               = "1.0"
+      retention_policy_days = 7
+    }
+  }
+}
+
+resource "azurerm_storage_queue" "main" {
+  count                = var.to_provision == true ? 1 : 0
+  name                 = "funcqueue"
+  storage_account_name = azurerm_storage_account.func[count.index].name
+}
+
+resource "azurerm_storage_container" "deployments" {
+  count                 = var.to_provision == true ? 1 : 0
+  name                  = "billing-alert-function-releases"
+  storage_account_name  = azurerm_storage_account.func[count.index].name
+  container_access_type = "private"
+}
+
+resource "azurerm_storage_blob" "storage_blob" {
+  count                  = var.to_provision == true ? 1 : 0
+  name                   = "billing-alert-function-blob"
+  storage_account_name   = azurerm_storage_account.func[count.index].name
+  storage_container_name = azurerm_storage_container.deployments[count.index].name
+  type                   = "Block"
+}
+
+resource "azurerm_app_service_plan" "main" {
+  count               = var.to_provision == true ? 1 : 0
+  name                = "plan-${var.func_identifier}"
+  location            = azurerm_resource_group.main[count.index].location
+  resource_group_name = azurerm_resource_group.main[count.index].name
+  kind                = "Linux"
+  reserved            = true
+
+  sku {
+    tier = var.app_service_plan.tier
+    size = var.app_service_plan.size
+  }
+}
+
+resource "azurerm_application_insights" "main" {
+  count               = var.to_provision == true ? 1 : 0
+  name                = "appi-${var.func_identifier}"
+  location            = azurerm_resource_group.main[count.index].location
+  resource_group_name = azurerm_resource_group.main[count.index].name
+  application_type    = "Node.JS"
+  tags                = {}
```

Comment:
It might be a good idea to take in a `tags` variable that we can apply to all resources that support tags.

Reply:
Again, great idea. Implementing it now